astropy/io/fits/hdu/image.py

```diff
@@ -214,6 +215,149 @@ def header(self, header):
         self._modified = True
         self.update_header()
 
+    @property
+    def subset(self):
+        """Read a subset of the image into a `~numpy.ndarray`.
+
+        This property enables a subset of the FITS image to be obtained without
+        reading or downloading the entire file.  This is achieved by dynamically
+        translating data slicing operations into a set of file seek and read
+        operations which only access the necessary "chunks" of the file.
+
+        The number of "chunks" in a file is defined as the length of the first
+        dimension.  For example, an image of shape (100, 200) is defined to be
+        composed of 100 chunks.  As a result, slicing operations which limit
+        the data required along the first dimension will be more efficient.
+        For example, `.subset[10:12, :]` would trigger two chunks to be
```

Comment:
these links will fail the CI, I would suggest running the tests and build the docs on this locally before you open PR.

Reply:
Agreed!  Will do.

Reply:
This should be resolved now -- I attempted to add extensive docstrings and narrative docs!